x/auth/ante.go

```diff
@@ -176,31 +175,46 @@ func processSig(
 		// Handle w/ #870
 		panic(err)
 	}
+	// If pubkey is not known for account,
+	// set it from the StdSignature.
+	pubKey, res := processPubKey(acc, sig, simulate)
+	if !res.IsOK() {
+		return nil, res
+	}
+	err = acc.SetPubKey(pubKey)
```

Comment:
The account may already have it's pubkey set, right? Maybe move the setting logic to `processPubKey`?

Reply:
> The account may already have it's pubkey set, right?
Not necessarily. `processPubKey` just processes the public key and returns it, IMO it shouldn't be its responsibility to set the account's respective field.

Reply:
Right, but the account may already have it set. Previously, if it was already set, it was a no-op. Now we always set. Not a big deal either way I suppose.

Reply:
I could drop change it as follows:

```diff
diff --git a/x/auth/ante.go b/x/auth/ante.go
index 94899a87..37d7636f 100644
--- a/x/auth/ante.go
+++ b/x/auth/ante.go
@@ -187,9 +187,11 @@ func processSig(
        if !res.IsOK() {
                return nil, res
        }
-       err = acc.SetPubKey(pubKey)
-       if err != nil {
-               return nil, sdk.ErrInternal("setting PubKey on signer's account").Result()
+       if acc.GetPubKey() == nil {
+               err = acc.SetPubKey(pubKey)
+               if err != nil {
+                       return nil, sdk.ErrInternal("setting PubKey on signer's account").Result()
+               }
        }
 
        consumeSignatureVerificationGas(ctx.GasMeter(), pubKey)
```

But it does not have any impact either way.

Reply:
OK 👌